### PR TITLE
ted override error handling TM-3239

### PR DESCRIPTION
**Pull request by SidelineCory24**
In short, the date "0001-01-01T00:00:00" is being returned by WS and is throwing `ValueError: year 0 is out of range`.

The datetime function is looking/handling a lot of things. It is expecting certain ranges to be met (ie: only 12 months so you can't have `13-13-13` as a date even though the year and day are valid, etc.). With that being the case, this PR attempts to create a valid date out of a real but illogical date like the error we're receiving, but if it can't create the date at all it will show `Invalid date`. 

To-Do:

- [x] Display `01/01/0001` when `0001-01-01T00:00:00` is returned
- [x] Displays `Invalid date` when an unreal date like `01-32-32T000:00:00` is given or it can't handle the date 

Edge cases to manually test in the PV services file (order can be rearranged ie: 01-0001-01/add any number ie: 01-23-1234):
- "0001-01-01T00:00:00"
- "001-01-01T00:00:00"
- "01-01-01T00:00:00"
- Whatever your ❤️ desirers

### CODE REVIEW

File: talentmap_api/common/common_helpers.py [hunk 1]
```diff
@@ -177,15 +177,18 @@ def ensure_date(date, utc_offset=0):
...
+            return None
+        elif isinstance(date, str):
+            return parser.parse(date).astimezone(datetime.timezone.utc) - datetime.timedelta(hours=utc_offset)
+        elif isinstance(date, datetime.date):
+            return date.astimezone(datetime.timezone(datetime.timedelta(hours=utc_offset)))
+        else:
+            logger.warn(f"Parameter {date} must be a date object or string")
+            return None
+    except ValueError:
+        return "Invalid date"
```
Comment (SidelineCory24) [inline talentmap_api/common/common_helpers.py:191]: Didn't want to use two try/except so instead returning `Invalid date` because there two different Value errors, a bad data `0001-01-01` and an invalid date `2022-32-01`

```
try:
        date = date + '.000Z'
        return parser.parse(date).astimezone(datetime.timezone.utc) - datetime.timedelta(hours=utc_offset)
except:
        return "Invalid date"
```

**Comment by SidelineCory24** [comment]
I prefer to merge this PR once I get the list from Pat of the positions in DEV1 that have this error. It appears that the positions are not showing in TMAP (but are in swagger) at all if they have this bad date in DEV1 and I want to cross reference that with Pat's list to be safe.

**Comment by scott062** [comment]
image
This was modifying the override directly in the mock (postgres), still testing the code though. Best test will be in dev1 however

**Review by scott062 — APPROVED** [review]
⌚ 📆
